src/cmd/ksh93/features/poll

```diff
@@ -4,7 +4,7 @@ lib	select,poll,socket
 lib	htons,htonl sys/types.h sys/socket.h netinet/in.h
 lib	getaddrinfo sys/types.h sys/socket.h netdb.h
 typ	fd_set sys/socket.h sys/select.h
-tst	pipe_socketpair note{ use socketpair() for peekable pipe() }end execute{
+tst	pipe_socketpair note{ use socketpair() for peekable pipe() }end compile{
```

Comment:
Can we assume pipes are peekable on all modern OS's ?

Reply:
What does "peekable" mean? This appears to be mostly relevant to the *src/lib/libast/sfio/sfpkrd.c* module. Specifically, the `sfpkrd()` function. Looking at the code it is clear it involves a feature only available on SysVR4 based systems (like Solaris and DYNIX/ptx) that implement the STREAMS subsystem.

I have no idea if this change is safe let alone correct. This is another example where the AST code is an impenetrable mess of support for platforms we probably no longer care about. Or at least, in the case of Solaris, might be willing to support as second class distros with possibly suboptimal behavior.

Reply:
It is enabled on Linux so definitely it's not specific to Solaris.

Reply:
Yes, but note that it is predicated on symbol `_stream_peek`. See *src/cmd/ksh93/sh/io.c* where you'll find lines like

```
if _pipe_socketpair && !_stream_peek
```

Pipes are definitely not "peekable" on any system other than one derived from SysVR4 with support for the STREAMS subsystem. Being "peekable" means it supports `ioctl(I_PEEK)`. I can't easily tell what this test is actually trying to determine.

Note that documents like https://linux.die.net/man/3/ioctl leave me very confused as I do not believe that Linux supports STREAMS devices. At least not without 3rd-party extensions. See https://en.wikipedia.org/wiki/STREAMS.

Reply:
Given the ambiguity of the code it might be correct that `socketpair()` based pipes are "peekable" in the sense meant by ksh. But it looks to me like the ksh concept of "peekable" definitely is not true for pipes on any system other than one derived from SysVR4 which implements STREAM pipes.

Reply:
This may be a situation where we should just make the change and deal with any problems down the road. Your call, @siteshwar, whether the risk is worth it.

Reply:
Let's just ship it.

Reply:
Fine by me. I'll merge this as soon as I verify it doesn't break building on my local systems.

Reply:
Note that my recent change to emit a backtrace upon receipt of SIGSEGV broken the *bin/package* based build on FreeBSD.  So I can't verify this change on that platform without using `meson`. I'm reluctant to figure out how to add the relevant test for `-lexecinfo` to the Nmake based build since it seems like we are really close to being able to use Meson for builds locally and on Travis.